File: sortingAlgorithms/Python/BubbleSort.py
def BubbleSort(array):
    length = len(array)
    temp = 0
    for i in range(length - 1):
        for j in range(0,length - i - 1):
            if array[j] > array[j+1]:
                temp = array[j+1]
                array[j+1] = array[j]
                array[j] = temp


array = [3,2,8,4,7]
print("Unsorted Array:")
print(array)
BubbleSort(array)
print("Sorted Array")
print(array)

File: sortingAlgorithms/Java/BubbleSort.java
package sortingAlgorithms.Java;
public class BubbleSort {
    private static void bubbleSort(int[] array){
        int length = array.length;
        int temp = 0;

        for (int i = 0; i < length; i++){
            for(int j = 1; j < (length-i); j++){
                if(array[j-1] > array[j]) {
                    //swap the elements
                    temp = array[j-1];
                    array[j-1] = array[j];
                    array[j] = temp;
                }
            }
        }
    }
    public static void main(String[] args){
        int[] array = {2,4,1,8,6, 7 , 90, 70, 20, 50};
        bubbleSort(array);
        for (int i = 0; i < array.length; i++){
            System.out.println(array[i]);
        }
    }
}